curriculum/challenges/english/25-front-end-development/lecture-searching-and-sorting-algorithms/68420c70e4bfe26d52b780dc.md

```diff
@@ -1,135 +1,189 @@
 ---
 id: 68420c70e4bfe26d52b780dc
-# title needs to be updated to correct title when lectures are finalized
-title: Searching and Sorting Algorithms
+title: What Is Binary Search and How Does It Differ From Linear Search?
 challengeType: 19
-# dashedName needs to be updated to correct title when lectures are finalized
-dashedName: lecture-searching-and-sorting-algorithms
+dashedName: what-is-binary-search-and-how-does-it-differ-from-linear-search
 ---
 
 # --description--
 
-Watch the video or read the transcript and answer the questions below.
+Searching through a list of items is a common occurrence in computer science. There are two key algorithms you should know about when it comes to searching: linear search and binary search.
+
+Linear search starts at the beginning of a list and iterates through each item until it finds the target value it is looking for.
+
+If the target value is found, the index where it's located in the list is returned. If the target value isn't found, `-1` is returned. We return `-1` because it's not a valid index in most programming languages.
+
+Here is what the code looks like for linear search:
+
+```python
+def linear_search(arr, target):
+    for i in range(len(arr)):
+        if arr[i] == target:
+            return i
+    return -1
+```
+
+If the list we'll search through is `[13, 4, 7, 9, 10]` and the target value is `9`, the function would return `3` because `9` is at index `3`.
+
+If we changed the target value to `5`, the function would return `-1` because `5` is not in the list.
+
+While this is a relatively straight forward algorithm, it is not the most efficient. If you have a large list of items, linear search can take a long time to find the target value.
+
+The time complexity of linear search is O(n) because the time it takes to search through the list grows linearly with the size of the list.
```

Comment:
```suggestion
The time complexity of linear search is `O(n)` because the time it takes to search through the list grows linearly with the size of the list.
```